android/expoview/src/main/java/versioned/host/exp/exponent/modules/api/ImageManipulatorModule.java

```diff
@@ -42,32 +47,37 @@ public String getName() {
   }
 
   @ReactMethod
-  public void manipulate(String uri, ReadableArray actions, ReadableMap saveOptions, Promise promise) {
-    String decoded = Uri.decode(uri);
-    Bitmap bmp = null;
-    try {
-      bmp = ImageLoader.getInstance().loadImageSync(decoded,
-          new DisplayImageOptions.Builder()
-              .cacheOnDisk(true)
-              .imageScaleType(ImageScaleType.NONE)
-              .considerExifParams(true)
-              .build());
-    } catch (Throwable e) {}
-    if (bmp == null) {
-      try {
-        bmp = ImageLoader.getInstance().loadImageSync(uri,
-            new DisplayImageOptions.Builder()
-                .cacheOnDisk(true)
-                .imageScaleType(ImageScaleType.NONE)
-                .considerExifParams(true)
-                .build());
-      } catch (Throwable e) {}
-    }
-    if (bmp == null) {
-      promise.reject(new IllegalStateException("Opening the image failed."));
-      return;
-    }
+  public void manipulate(final String uri, final ReadableArray actions, final ReadableMap saveOptions, final Promise promise) {
+    ImageRequest imageRequest = ImageRequest.fromUri(uri);
```

Comment:
Probably we want to use `.setRotationOptions(RotationOptions.autoRotate())` option here.
[link](https://frescolib.org/docs/rotation.html)